pkg/integrations/redis_exporter/redis_exporter.go

```diff
@@ -0,0 +1,149 @@
+// package redis_exporter embeds https://github.com/oliver006/redis_exporter
+package redis_exporter //nolint:golint
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+
+	re "github.com/oliver006/redis_exporter/lib/exporter"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+// Integration is the redis_exporter integration. The integration queries
+// a redis instance's INFO and exposes the results as metrics.
+type Integration struct {
+	c        Config
+	exporter *re.Exporter
+}
+
+// New creates a new redis_exporter integration.
+func New(log log.Logger, c Config) (*Integration, error) {
+	level.Debug(log).Log("msg", "initialising redis_exporer with config %v", c)
+
+	address := c.RedisAddr
+	if len(address) == 0 {
+		address = os.Getenv("REDIS_EXPORTER_ADDRESS")
+	}
+	if len(address) == 0 {
+		return nil, fmt.Errorf("cannot create redis_exporter; neither redis_exporter.redis_addr or $REDIS_EXPORTER_ADDRESS is set")
+	}
+	level.Debug(log).Log("msg", "Redis exporter address is %s", address)
+
+	exporterConfig := c.GetExporterOptions()
+
+	if c.ScriptPath != "" {
+		ls, err := ioutil.ReadFile(c.ScriptPath)
+		if err != nil {
+			return nil, fmt.Errorf("Error loading script file %s    err: %s", c.ScriptPath, err)
```

Comment:
Use the standard go error wrapping here: 

```suggestion
			return nil, fmt.Errorf("Error loading script file %s: %w", c.ScriptPath, err)
```